aws/scenarios/microVMs/microVMs/libvirt.go

```diff
@@ -0,0 +1,185 @@
+package microVMs
+
+import (
+	"crypto/rand"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/DataDog/test-infra-definitions/aws/scenarios/microVMs/vmconfig"
+	"github.com/DataDog/test-infra-definitions/command"
+	"github.com/pulumi/pulumi-libvirt/sdk/go/libvirt"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+const (
+	dhcpEntriesTemplate   = "<host mac='%s' name='%s' ip='%s'/>"
+	microVMGroupSubnet    = "169.254.0.0/16"
+	domainSocketCreateCmd = `rm -f /tmp/%s.sock && python3 -c "import socket as s; sock = s.socket(s.AF_UNIX); sock.bind('/tmp/%s.sock')"`
+)
+
+var subnetGroupMask = net.IPv4Mask(255, 255, 255, 0)
+
+type DomainTemplate struct {
+	Mac string
+}
+
+type DomainParameters struct {
+	filesystemID pulumi.IDOutput
+	ip           string
+	xls          string
+	domainName   string
+}
+
+func getNextVMSubnet(ip net.IP) net.IP {
+	ipv4 := ip.To4()
+	ipv4 = ipv4.Mask(subnetGroupMask)
+	ipv4[2]++
+
+	return ipv4
+}
+
+func generateNewUnicastMac() (string, error) {
+	buf := make([]byte, 6)
+	_, err := rand.Read(buf)
+	if err != nil {
+		return "", err
+	}
+
+	// Set LSB bit of MSB byte to 0
+	// This denotes unicast mac address
+	buf[0] &= 0xfe
+	return fmt.Sprintf("%02x:%02x:%02x:%02x:%02x:%02x", buf[0], buf[1], buf[2], buf[3], buf[4], buf[5]), nil
+}
+
+func setupLibvirtVM(ctx *pulumi.Context, runner *command.Runner, libvirtUri pulumi.StringOutput, vmset vmconfig.VMSet, waitForList []pulumi.Resource) error {
+	var domainDependencies []pulumi.Resource
+	baseVolumeId := generatePoolPath(vmset.Name) + basefsName
+
+	provider, err := libvirt.NewProvider(ctx, "provider", &libvirt.ProviderArgs{
+		Uri: libvirtUri,
+	}, pulumi.DependsOn(waitForList))
+	if err != nil {
+		return err
+	}
+
+	domainXls, err := os.ReadFile("resources/domain.xls")
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
You can embed directive to embed this file in the binary with `//go:embed` https://pkg.go.dev/embed
It raises compile time errors if the file is missing, no need to read the file content. Wrote a tiny comment about it at https://github.com/pducolin/advent-of-code#embed